modules/internal/ChapelBase.chpl

```diff
@@ -840,8 +841,14 @@ module ChapelBase {
   // locale as the sync/cofall/cobegin was initiated on and thus the
   // same locale on which the object is allocated.
   pragma "dont disable remote value forwarding"
-  inline proc _endCountAlloc() {
-    return new _EndCount();
+  inline proc _endCountAlloc(param forceLocalTypes : bool) {
```

Comment:
Also, it might be a little cleaner to have `forceLocalTypes` default to `false`, so you don't have to set it for every `_endCountAlloc`